src/Utils/Security.php

```diff
@@ -98,20 +100,8 @@ public static function castKey(XMLSecurityKey $key, string $algorithm, string $t
             return $key;
         }
 
-        if (
-            !in_array(
-                $algorithm,
-                [
-                    XMLSecurityKey::RSA_1_5,
-                    XMLSecurityKey::RSA_SHA1,
-                    XMLSecurityKey::RSA_SHA256,
-                    XMLSecurityKey::RSA_SHA384,
-                    XMLSecurityKey::RSA_SHA512
-                ],
-                true
-            )
-        ) {
-            throw new Exception('Unsupported signing algorithm.');
+        if (!in_array($algorithm, C::$RSA_DIGESTS, true)) {
```

Comment:
This one is wrong, I believe. Should be:

```php
        if (!in_array($algorithm, array_keys(C::$RSA_DIGESTS, true)) {
```

Reply:
Yes, thank you, lack of tests!